### DISCUSSION TITLE
Login works, but blank screen is shown after that

### DISCUSSION THREAD

**Discussion by dgard1981** [coder/code-server#6427] [Q&A]
Hey everyone,

I'm trying to use `code-server` with Ubuntu 22.04. All appeared to be working and I am able to see the login page and successfully login, but no matter how I configure `nginx`, all I get after a successful login is a blank screen. The relevant entries from the `nginx` error log can be found below, along with the `nginx` and `code-server` configurations.

Using dev-tools I have noted the following -
- On the login page, all assets are loaded from `/_static/src/browser`, and are successfully retrieved.
- After login, all assets fail to load from `stable-2ccd690cbff1569e4a83d7c43d45101f817401dc/static`, with two exceptions. In the case of both exceptions, the same asset from the same path loaded successfully on the login page.
   1.  `manifest.json` fails to load from `/manifest.json`.
   2 . `favicon.ico` fails to load from `_static/src/browser/media/favicon.ico`.

![login](https://github.com/coder/code-server/assets/43265354/b4f353ad-3e5a-4cce-9f1e-5b487724f6e3)
![after-login](https://github.com/coder/code-server/assets/43265354/60471de7-05d2-496a-85c9-ae92d8ce12b4)

I did initially wonder if the issue was due to a timeout, but I have tried explicitly adding timeouts that far exceed the load/error time.

Note that the resources yielding errors will load when requested through Postman, or even a browser. It only seems to be `nginx` that is having a problem.

I have searched for the files in question (such as `workbench.web.main.js`) and confirmed that all can be read globally (`-rw-r--r--`).

I have also inspected all logs at **~/.local/share/code-server/logs** and they are all empty.

Is anyone able to suggest what the problem may be?

---

### `code-server` config

Note that I am using HTTPS to connect, with an AWS EC2 instance behind an Application Load Balancer, where TLS terminates. However, I have also tested using HTTP to connect direct to the EC2 instance, and the issue remains.

I have checked to ensure that `code-server` is running using `sudo systemctl status code-server@$USER`, though it must be running otherwise I would see an error rather than the login page. According to the output, the HTTP server is listening on the expected port.

> info  HTTP server listening on http://127.0.0.1:8080/

```conf
bind-addr: 127.0.0.1:8080
auth: password
password: somepassword
cert: false
```

### `nginx` error log.

```log
2023/09/12 09:26:28 [error] 1956#1956: *60 upstream prematurely closed connection while reading response header from upstream, client: xxx.xxx.xxx.xxx, server: code.mydomain.com, request: "GET /stable-2ccd690cbff1569e4a83d7c43d45101f817401dc/static/out/vs/code/browser/workbench/workbench.js HTTP/1.1", upstream: "http://127.0.0.1:8080/stable-2ccd690cbff1569e4a83d7c43d45101f817401dc/static/out/vs/code/browser/workbench/workbench.js", host: "code.mydomain.com", referrer: "http://code.mydomain.com/"
2023/09/12 09:26:28 [error] 1955#1955: *50 upstream prematurely closed connection while reading response header from upstream, client: xxx.xxx.xxx.xxx, server: code.mydomain.com, request: "GET /stable-2ccd690cbff1569e4a83d7c43d45101f817401dc/static/out/vs/loader.js HTTP/1.1", upstream: "http://127.0.0.1:8080/stable-2ccd690cbff1569e4a83d7c43d45101f817401dc/static/out/vs/loader.js", host: "code.mydomain.com", referrer: "http://code.mydomain.com/"
2023/09/12 09:26:28 [error] 1956#1956: *57 upstream prematurely closed connection while reading response header from upstream, client: xxx.xxx.xxx.xxx, server: code.mydomain.com, request: "GET /stable-2ccd690cbff1569e4a83d7c43d45101f817401dc/static/out/vs/workbench/workbench.web.main.js HTTP/1.1", upstream: "http://127.0.0.1:8080/stable-2ccd690cbff1569e4a83d7c43d45101f817401dc/static/out/vs/workbench/workbench.web.main.js", host: "code.mydomain.com", referrer: "http://code.mydomain.com/"
2023/09/12 09:26:28 [error] 1956#1956: *55 upstream prematurely closed connection while reading response header from upstream, client: xxx.xxx.xxx.xxx, server: code.mydomain.com, request: "GET /stable-2ccd690cbff1569e4a83d7c43d45101f817401dc/static/out/vs/workbench/workbench.web.main.nls.js HTTP/1.1", upstream: "http://127.0.0.1:8080/stable-2ccd690cbff1569e4a83d7c43d45101f817401dc/static/out/vs/workbench/workbench.web.main.nls.js", host: "code.mydomain.com", referrer: "http://code.mydomain.com/"
2023/09/12 09:26:28 [error] 1956#1956: *54 upstream prematurely closed connection while reading response header from upstream, client: xxx.xxx.xxx.xxx, server: code.mydomain.com, request: "GET /stable-2ccd690cbff1569e4a83d7c43d45101f817401dc/static/out/vs/webPackagePaths.js HTTP/1.1", upstream: "http://127.0.0.1:8080/stable-2ccd690cbff1569e4a83d7c43d45101f817401dc/static/out/vs/webPackagePaths.js", host: "code.mydomain.com", referrer: "http://code.mydomain.com/"
2023/09/12 09:26:28 [error] 1956#1956: *49 upstream prematurely closed connection while reading response header from upstream, client: xxx.xxx.xxx.xxx, server: code.mydomain.com, request: "GET /stable-2ccd690cbff1569e4a83d7c43d45101f817401dc/static/out/vs/workbench/workbench.web.main.css HTTP/1.1", upstream: "http://127.0.0.1:8080/stable-2ccd690cbff1569e4a83d7c43d45101f817401dc/static/out/vs/workbench/workbench.web.main.css", host: "code.mydomain.com", referrer: "http://code.mydomain.com/"
2023/09/12 09:26:28 [error] 1956#1956: *49 connect() failed (111: Unknown error) while connecting to upstream, client: xxx.xxx.xxx.xxx, server: code.mydomain.com, request: "GET /_static/src/browser/media/favicon.ico HTTP/1.1", upstream: "http://127.0.0.1:8080/_static/src/browser/media/favicon.ico", host: "code.mydomain.com", referrer: "http://code.mydomain.com/"
2023/09/12 09:26:28 [error] 1956#1956: *49 connect() failed (111: Unknown error) while connecting to upstream, client: xxx.xxx.xxx.xxx, server: code.mydomain.com, request: "GET /_static/src/browser/media/favicon-dark-support.svg HTTP/1.1", upstream: "http://127.0.0.1:8080/_static/src/browser/media/favicon-dark-support.svg", host: "code.mydomain.com", referrer: "http://code.mydomain.com/"
```

### `nginx` config at **/etc/nginx/sites-available/code-server**

```conf
server {
    listen 80;
    listen [::]:80;
    server_name code.mydomain.com;

    location / {
        proxy_pass http://127.0.0.1:8080/;
        proxy_set_header Host $host;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection upgrade;
        proxy_set_header Accept-Encoding gzip;
        proxy_read_timeout 60s;
        proxy_connect_timeout 30s;
    }
}
```

### `nginx` config at **/etc/nginx/nginx.conf**

```conf
user www-data;
worker_processes auto;
pid /run/nginx.pid;
include /etc/nginx/modules-enabled/*.conf;

events {
        worker_connections 768;
}

http {
    sendfile on;
    tcp_nopush on;
    types_hash_max_size 2048;
    server_names_hash_bucket_size 128;

    include /etc/nginx/mime.types;
    default_type application/octet-stream;

    ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3;
    ssl_prefer_server_ciphers on;

    access_log /var/log/nginx/access.log;
    error_log /var/log/nginx/error.log;

    gzip on;

    include /etc/nginx/conf.d/*.conf;
    include /etc/nginx/sites-enabled/*;
}
```

**Comment by dgard1981** [comment]
The issue here seems to be the password authentication. If I set that to `none`, then `nginx` serves up code-server as expected.

Obviously though, I don't want to run code-server without any protection. Is this a known issue? I can find many references to code-server failing to serve a page after login, but those all seem to be related to HTTPS. However, I experience the issue using both HTTP (direct to the EC2 instance) and HTTPS (via an Application Load Balancer).

**Reply by code-asher** [reply]
Nope, not a known issue.  Do you have code-server's stdout?  It should also be in `~/.local/share/code-server/coder-logs`, I think.
